test/TestNestedJinjaRule.py

```diff
@@ -0,0 +1,209 @@
+  tasks:
+    - name: multiline non-nested example
+      set_fact:
+        var_one_ml: >
+          number for 'one' is {{
+          2 * 1 }}
+''')
+
+SUCCESS_TASK_2P = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: nesting far from each other
+      set_fact:
+        var_two: "number for 'two' is {{ 2 * 1 }} and number for 'three' is {{ 4 - 1 }}"
+''')
+
+SUCCESS_TASK_2P_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: multiline nesting far from each other
+      set_fact:
+        var_two_ml: >
+          number for 'two' is {{ 2 * 1
+          }} and number for 'three' is {{
+          4 - 1 }}
+''')
+
+SUCCESS_TASK_C_2P = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: nesting close to each other
+      set_fact:
+        var_three: "number for 'ten' is {{ 2 - 1 }}{{ 3 - 3 }}"
+''')
+
+SUCCESS_TASK_C_2P_M = PlayFile('playbook.yml', '''
+- hosts: all
+  tasks:
+    - name: multiline nesting close to each other
+      set_fact:
+        var_three_ml: >
+          number for 'ten' is {{
+          2 - 1
+          }}{{ 3 - 3 }}
+''')
+
+
+@pytest.fixture
+def runner(tmp_path, default_rules_collection):
+    return Runner(
+        default_rules_collection,
+        str(tmp_path / 'playbook.yml'),
+        [], [], [],
+    )
+
+
+@pytest.fixture
+def _playbook_file(tmp_path, request):
+    if request.param is None:
+        return
+    for play_file in request.param:
+        p = tmp_path / play_file.name
+        p.write_text(play_file.content)
+
+
+@pytest.mark.parametrize(
+    '_playbook_file',
+    (
+        pytest.param([FAIL_TASK_1LN], id='file includes one-level nesting'),
+        pytest.param([FAIL_TASK_1LN_M], id='file includes one-level multiline nesting'),
+        pytest.param([FAIL_TASK_2LN], id='file includes two-level nesting'),
+        pytest.param([FAIL_TASK_2LN_M], id='file includes two-level multiline nesting'),
+        pytest.param([FAIL_TASK_W_5LN], id='file includes five-level wild nesting'),
+        pytest.param([FAIL_TASK_W_5LN_M], id='file includes five-level wild multiline nesting'),
+    ),
+    indirect=['_playbook_file'],
+)
+@pytest.mark.usefixtures('_playbook_file')
+def test_including_wrong_nested_jinja(runner):
+    results = runner.run()
+    assert results[0].rule.shortdesc == 'Nested jinja pattern'
+
+
+@pytest.mark.parametrize(
+    '_playbook_file',
+    (
+        pytest.param([SUCCESS_TASK_P], id='file includes non-nested example'),
+        pytest.param([SUCCESS_TASK_P_M], id='file includes multiline non-nested example'),
+        pytest.param([SUCCESS_TASK_2P], id='file includes nesting far from each other'),
+        pytest.param([SUCCESS_TASK_2P_M], id='file includes multiline nesting far from each other'),
+        pytest.param([SUCCESS_TASK_C_2P], id='file includes nesting close to each other'),
+        pytest.param(
+            [SUCCESS_TASK_C_2P_M],
+            id='file includes multiline nesting close to each other'
```

Comment:
Nitpick: let's have a trailing comma here too, for consistency.
```suggestion
            id='file includes multiline nesting close to each other',
```